draft-ietf-tls-svcb-ech.md

```diff
@@ -90,12 +90,14 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
 
 If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MUST adopt SVCB-reliant behavior (as in {{disabling-fallback}}) for that RRSet.  This precludes the use of certain connections that Alt-Svc would otherwise allow, as discussed in {{Section 9.3 of !SVCB}}.
 
-## Security Considerations
+# Security Considerations
 
 A SVCB RRSet containing some RRs with "ech" and some without is vulnerable to a downgrade attack: a network intermediary can block connections to the endpoints that support ECH, causing the client to fall back to a non-ECH endpoint.  This configuration is NOT RECOMMENDED. Zone owners who do use such a mixed configuration SHOULD mark the RRs with "ech" as more preferred (i.e. lower SvcPriority value) than those without, in order to maximize the likelihood that ECH will be used in the absence of an active adversary.
 
 In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
 
+The name of the destination domain is protected from disclosure by ECH, but it is still revealed by the DNS queries used to reach the server.  To achieve the security goals from {{Section 10.1 of !ECH}}, clients for this specification SHOULD select a DNS resolver that they trust to preserve the confidentiality of their queries and return authentic answers, and use authenticated encryption for DNS queries and responses.
```

Comment:
looks good to me.

Reply:
OK, I've incorporated a version of this wording.